sdks/go/pkg/beam/io/fhirio/search.go

```
@@ -0,0 +1,167 @@
+		return fn.searchResources(query)
+	})
+	if err != nil {
+		fn.resourcesErrorCount.Inc(ctx, 1)
+		emitDeadLetter(errors.Wrapf(err, "error occurred while performing search for query: [%v]", query).Error())
+		return
+	}
+
+	fn.resourcesSuccessCount.Inc(ctx, 1)
+	emitFoundResources(query.Identifier, resourcesFound)
+}
+
+func (fn *searchResourcesFn) searchResources(query SearchQuery) ([]string, error) {
+	allResources := make([]string, 0)
+	resourcesInPage, nextPageToken, err := fn.searchResourcesPaginated(query, "")
+	for nextPageToken != "" {
+		allResources = append(allResources, resourcesInPage...)
+		resourcesInPage, nextPageToken, err = fn.searchResourcesPaginated(query, nextPageToken)
+	}
+	// Must add resources from last page.
+	allResources = append(allResources, resourcesInPage...)
+	return allResources, err
+}
+
+// Performs a search request retrieving results only from the page identified by
+// `pageToken`. If `pageToken` is the empty string it will retrieve the results
+// from the first page.
+func (fn *searchResourcesFn) searchResourcesPaginated(query SearchQuery, pageToken string) ([]string, string, error) {
+	response, err := fn.client.search(fn.FhirStorePath, query.ResourceType, query.Parameters, pageToken)
+	if err != nil {
+		return nil, "", err
+	}
+
+	body, err := extractBodyFrom(response)
+	if err != nil {
+		return nil, "", err
+	}
+
+	var bodyFields struct {
+		Entries []interface{}        `json:"entry"`
+		Links   []responseLinkFields `json:"link"`
+	}
+	err = json.NewDecoder(strings.NewReader(body)).Decode(&bodyFields)
+	if err != nil {
+		return nil, "", err
+	}
+
+	resourcesFoundInPage := mapEntryToString(bodyFields.Entries)
+	return resourcesFoundInPage, extractNextPageTokenFrom(bodyFields.Links), nil
+}
+
+func mapEntryToString(entries []interface{}) []string {
+	stringifiedEntries := make([]string, 0)
+	for _, entry := range entries {
+		entryBytes, err := json.Marshal(entry)
+		if err != nil {
+			continue
```

Hmm we should at least log an error here somehow, if we silently drop an entry.